SmartPlaces.Facilities/samples/Topology/README.md

```diff
@@ -49,3 +49,102 @@ The following secrets must be defined in the KeyVault in order to run this code:
+        4. Number of Relationship updates succeeded by RelationshipType
+        5. Number of Relationship updates failed by RelationshipType
+        6. Number of Throttled Relationship updates by RelationshipType
+        7. Number of Relationship updates unchanged by RelationshipType (due to no differences between source and target)
+    4. Ontology Mapping Metrics
+        1. Mapping for Input DTMI not found: 
+            - Counts the number of the Incoming DTMIs that do not exist in the Source Ontology file that is loaded. 
+            - Entries here indicate that the topology program assemblies are out of sync with the source system's ontology
+        2. Output Mapping for Input DTMI not found:
+            - Counts the number of the Incoming DTMIs that are not mapped to output DTMIs in the mapping file
+            - Entries here indicate that the ontology mapping from source to target is incomplete and should be updated. It is possible that the target ontology is a smaller set than the
+              input ontology and some mappings may not be possible (or needed).
+        3. Target DTMI not found:
+            - An target dtmi entry exists in the mapping file which does not actually exist in the target ontology
+        4. InputInterfaceNotFound not found in Model by InterfaceType
+            - A source dtmi entry exists in the mapping file which does not actually exist in the source ontology
+        5. Relationship Not Found in Model by RelationshipType
+            - A relationship was found in the toplogy that is not found in the target ontology
+        6. Duplicate property found in Model by PropertyName
+            - A invalid model was found with multiple properties with the same name
+        7. Invalid Target DTMI Mappings in MappingFile
+            - The mapping file has a dtmi that is not in the target ontology
+        8. Invalid Output DTMI by OutputDtmi
+            - The mapping file has a malformed DTMI
+
+        For any of of these issues which involve changes to the provided mapping files, you will need contribution privileges to the Azure\opendigitaltwins-tool repository to resolve. If you do not have permissions to contribute to this repo, please open an issue [here](https://github.com/Azure/opendigitaltwins-tools/issues)
+
+2. You can look at application logs directly through your Application Insights logs with the following queries:
+
+```
+
+traces
+| where customDimensions.CategoryName startswith 'Microsoft.SmartPlaces.Facilities.'
+
+```
+
+```
+
+exceptions
+| where customDimensions.CategoryName startswith 'Microsoft.SmartPlaces.Facilities.'
+
+```
+
+If you have set up logging for your container instance and routed the logs to a log analytics instance, you can look at the CustomLogs / ContainerInstanceLog_CL in your Log Analytics workspace. If your instance name is "topology", you can run the following query to get all of the logs 
```

Comment:
```markdown
3. If you have deployed this sample code within an Azure Container Instance there are additional methods for accessing the logs.
    - Within the AzureContainerInstance resource in the Azure Portal you can click `Containers` > `Logs` to view the last few minutes of console logs from the Topology process. 
    - If you've connected the AzureContainerInstance to a LogAnalytics workspace you can query the same console logs given the following queries.
    
    \```
    // Get all console logs
    ContainerInstanceLog_CL
    | where ContainerName_s contains("topology")
    | order by TimeGenerated asc
    
    //Filter for exceptions
    ContainerInstanceLog_CL
    | where ContainerName_s contains("topology")
    | where Message contains("exception")
    | order by TimeGenerated asc
    
    //Monitor progress
    ContainerInstanceLog_CL
    | where ContainerName_s contains("topology")
    | where Message contains("total")
    | order by TimeGenerated asc
    \```
    
    
 ```